packages/piqi/piqi.0.7.8/opam

```diff
@@ -0,0 +1,39 @@
+opam-version: "2.0"
+synopsis: "Protocol Buffers, JSON and XML serialization system for OCaml"
+authors: "Anton Lavrik <alavrik@piqi.org>"
+maintainer: "Anton Lavrik <alavrik@piqi.org>"
+license: "Apache-2.0"
+homepage: "https://github.com/alavrik/piqi-ocaml"
+bug-reports: "https://github.com/alavrik/piqi-ocaml/issues"
+depends: [
+  "ocaml" {>= "4.02.0"}
+  "dune"  {>= "2.0.0"}
+  "piqilib"
+  "stdlib-shims"
+  "num" {with-test}
+]
+dev-repo: "git+https://github.com/alavrik/piqi-ocaml"
+
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    "piqi,piqirun"
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+
+install: [
+  ["dune" "install" "-p" "piqi,piqirun"]
```

Comment:
if they are both installed in the same package, why are they different dune packages? It would simplify things a lot to just have one package or to separate piqi and piqirun in two separate packages if you prefer that way

Reply:
Because of the dune naming constraints. 
1) We need to install `piqirun.pb` and `piqirun.ext`, because these are the names that are used in downstream packages to link with the runtime. Dune requires libraries that all public libraries have a name that starts with a package name. So we had to add the `piqilib` package.

2) The piqi package contains the piqi compiler itself, and we can't put into the same piqilib package, because dune or opam will complain that the released package named and the package in the project differ.


And yes, splitting the opam `piqi` package into `piqirun` and `piqi` should be possible, and I think that in the longterm it would be a good idea to do this. But this is @alavrik's call, I just tried to switch to dune with minimal disruption. Basically, I tried to replicate the exact behavior of the current packages.

Reply:
We can wait for @alavrik's opinion. If he agrees to add an extra `piqirun` package to the piqi universe, I will happily split it. The `piqi` package will depend on `piqirun` so it will transitively bring `piqirun` and we won't break anything downstream. And the dune files will be so simplified that we could in the future just do `opam-release`.